Biblioteca Nacional de Maestros.js

```diff
@@ -0,0 +1,210 @@
+{
+	"translatorID": "b383df35-15e7-43ee-acd9-88fd62669083",
+	"label": "Biblioteca Nacional de Maestros",
+	"creator": "Sebastian Karcher",
+	"target": "^https?://www\\.bnm\\.me\\.gov\\.ar/catalogo",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2020-06-21 18:50:14"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2020 Sebastian Karcher
+	
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+
+function detectWeb(doc, url) {
+	// We'll go with Book throughout for simplicity's sake
+	if (/\/Record\/\d+/.test(url)) {
+		return "book";
+	}
+	else if (getSearchResults(doc, true)) {
+		return "multiple";
+	}
+	return false;
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('.result div[class*="resultItemLine"]>a.title[href*="/Record/"]');
+	for (let row of rows) {
+		let href = row.href;
+		let title = ZU.trimInternal(row.textContent);
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (items) scrape(Object.keys(items));
+		});
+	}
+	else {
+		scrape([url]);
+	}
+}
+
+
+function constructMARCurl(url) {
+	url = url.replace(/\/(Details|Holdings)([#?].*)?/, ""); // remove panels
+	return url + "/Export?style=MARCXML";
+}
+
+function scrape(urls) {
+	for (let url of urls) {
+	// Z.debug(url);
+	// eslint-disable-next-line loopfunc
```

Comment:
@dstillman just making sure disabling this rule is OK in this context? My understanding is that it's intended to avoid re-defining a function n times, which this doesn't do, but wasn't 100% sure (the reason there's a loop here is to avoid unnecessarily downloading the catalog item page from search results.)

Reply:
Did you mean `no-loop-func`? `loopfunc` is from JSHint, I think.

This is OK — the function doesn't reference any external variables, so there's nothing to get confused by here — but it does technically create multiple anonymous functions for no reason. You could just define `var handler = function (text) { … };` before the loop and pass `handler` to each invocation.